PocketKit/Sources/Sync/PocketSource.swift

```diff
@@ -504,6 +504,56 @@ extension PocketSource {
         }
     }
 
+    public func addHighlight(itemIID: NSManagedObjectID, patch: String, quote: String) {
+        space.performAndWait {
+            guard let savedItem = space.backgroundObject(with: itemIID) as? SavedItem, let context = savedItem.managedObjectContext, let itemID = savedItem.remoteID else {
+                Log.capture(message: "Could not retreive Saved Item to add highlight from background context for mutation")
+                return
+            }
+
+            let remoteID = UUID().uuidString
+
+            let highlight = Highlight(
+                context: context,
+                remoteID: remoteID,
+                createdAt: Date(),
+                updatedAt: Date(),
+                patch: patch,
+                quote: quote,
+                version: 2
+            )
+            var highlights = (savedItem.highlights?.array ?? [])
+            highlights.append(highlight)
+            savedItem.highlights = NSOrderedSet(array: highlights)
+
+            do {
+                try space.save()
+            } catch {
+                Log.capture(error: error)
```

Comment:
Do we need to do anything else here if this fails?

Reply:
Probably not at this stage, we use this pattern on almost all `Source` methods. We can consider revisiting (and perhaps consider if we still want to trigger backend updates or not), but I'd leave that to a broader `Source` revisit/refactor